utils/network/backend/redis/redis-storage.ts

```diff
@@ -3,59 +3,71 @@ import { createClient } from 'redis';
 import { logWarningInSentry } from '#utils/sentry';
 import { promiseTimeout } from './promise-timeout';
 
-let redisClient = createClient({
-  url: process.env.REDIS_URL,
-  pingInterval: 1000,
-});
-
-redisClient.on('error', (err) => {
-  logWarningInSentry('Error in Redis', { details: err });
-});
-
-export async function connect() {
-  if (!redisClient.isOpen) {
-    try {
-      return redisClient.connect();
-    } catch (e) {
-      logWarningInSentry('Could not connect to redis client');
+export class RedisClient {
+  static isRedisEnabled = process.env.REDIS_ENABLED === 'enabled';
+  private _client;
+
+  constructor(private cache_timeout: number) {
+    if (!RedisClient.isRedisEnabled) {
+      throw new Error('Redis is disabled');
     }
+
+    this._client = createClient({
+      url: process.env.REDIS_URL,
+      pingInterval: 1000,
+    });
+
+    this._client.on('error', (err) => {
+      logWarningInSentry('Error in Redis', { details: err });
+    });
   }
-}
 
-const redisStorage = (cache_timeout: number) =>
-  buildStorage({
-    async find(key: string) {
-      const result = await promiseTimeout(
-        redisClient.get(key),
-        100
-      ).catch((err) => {
+  connect = async () => {
+    if (!this._client.isOpen) {
+      try {
+        return this._client.connect();
+      } catch (e) {
+        logWarningInSentry('Could not connect to redis client');
+      }
+    }
+  };
+
+  find = async (key: string) => {
+    await this.connect();
+    console.log(key);
+
+    const result = await promiseTimeout(this._client.get(key), 100).catch(
+      (err) => {
         logWarningInSentry('Redis find key fail', { details: err });
         return null;
-      });
-
-      return result ? JSON.parse(result) : result;
-    },
-
-    async set(key: string, value: any) {
-      await promiseTimeout(
-        redisClient.set(key, JSON.stringify(value), {
-          PX: cache_timeout,
-        }),
-        200
-      ).catch((err) => {
-        logWarningInSentry('Redis set key failed', { details: err });
-      });
-    },
-
-    async remove(key: string) {
-      await redisClient.del(key);
-    },
-  });
+      }
+    );
 
-export default redisStorage;
-export async function disconnect() {
-  if (redisClient.isOpen) {
-    redisClient.removeAllListeners();
-    return await redisClient.disconnect();
-  }
+    return result ? JSON.parse(result) : result;
+  };
+
+  set = async (key: string, value: any) => {
```

Comment:
```suggestion
  async function set(key: string, value: any) {
```
Comme spécifié dans la [doc MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
> Arrow functions don't have their own [bindings](https://developer.mozilla.org/en-US/docs/Glossary/Binding) to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this), [arguments](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments), or [super](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super), and should not be used as [methods](https://developer.mozilla.org/en-US/docs/Glossary/Method).

De même pour les autres (remove & find)

Reply:
Alors je veux bien ton avis la dessus : c'est fait expres pour accéder au `this` de la classe et faire ensuite `this.connect` ou `this._client`

Reply:
Ta classe est déjà un `store`, tu n'as pas besoin de le redéfinir en public. Par conséquent : 
```
redisClient ? buildStorage(redisClient) : undefined,
```
devrait marcher et semble plus simple. Pas besoin de faire un bind dans un objet comme ça.

Reply:
...mais pour que ça marche, il faut utiliser des methods et pas des arrow function

(et désolé pour le français, je viens de me rendre compte :confused: )